### ISSUE TITLE
Remove authentication override

### ISSUE THREAD

**Issue by agrover** [mozilla/neqo#80]
```
// TODO(ekr@rtfm.com): IMPORTANT: This overrides
// authentication and so is fantastically dangerous.
// Fix before shipping.
```
@martinthomson Can you talk a little bit about the best timing of work to get this working properly, and what needs to be done, just so we all are clear?

**Comment by ddragana** [comment]
I am currently changing this. This is the firefox security stuff I am working on.
For firefox I am planning to have a connection event AuthenticationNeeded. The verification will be done in firefox code. When it is done Authenticated() will be called with success or error.
This is what will work for firefox, do we want something else for neqo without firefox?

**Comment by agrover** [comment]
> This is what will work for firefox, do we want something else for neqo without firefox?

If possible, yes.

**Comment by martinthomson** [comment]
The generic certificate authentication piece is a big lift.  Part of the longer term plans for NSS is to provide a generic certificate verifier based on what is in Firefox.  Right now, we only have the path building piece moved over, but we also want CRLite and other pieces integrated.  That's a very large project, and not really something that neqo can take on.

NSS does have certificate validation support (several in fact), but I would be reluctant to wire something like that up if there are plans to invalidate that work.  It's not exactly a trivial enterprise.  I would rather leave this unsolved here and leave users with the problem of validating the certificate chain (which we present through the `peer_certificate()` accessor).

**Comment by agrover** [comment]
OK. I just don't want to leave a loaded gun lying around for the Neqo-without-firefox case. We want to be able to test Neqo standalone but make it clear that if Neqo is integrated that cert auth must be provided.

Maybe there's a feature flag we can add for standalone use that allows testing, but also emits a scary message.

Tagging Future, we need something here but not right away.

**Comment by martinthomson** [comment]
With the changes to support gecko integration, I believe that this is fixed.  That is, our tests include the override, not the mainline code.  @ddragana, would you concur?

**Comment by ddragana** [comment]
Yes, this has been fixed.

**Comment by ddragana** [comment]
fixed by #106.
